FWCore/Framework/interface/stream/EDAnalyzerAdaptor.h

```diff
@@ -141,13 +142,14 @@ namespace edm {
         }
       }
 
-      void doBeginRun(RunPrincipal const& rp, EventSetupImpl const& ci, ModuleCallingContext const* mcc) final {
+      void doBeginRun(RunTransitionInfo const& info, ModuleCallingContext const* mcc) final {
+        RunPrincipal const& rp = info.principal();
```

Comment:
How about moving this inside the `if constexpr`? (there are many similar cases below, I won't repeat the comment) I'd expect the compiler to optimize it away in case the condition is false, but I think it would be tiny bit clearer as well.

Reply:
Done. That is better. Thanks. I also did the same thing in the other 7 places this occurs.